pkg/apis/networking/v1alpha1/clusteringress_types.go

```diff
@@ -94,15 +94,16 @@ type IngressSpec struct {
 	Rules []ClusterIngressRule `json:"rules,omitempty"`
 
 	// Visibility setting.
-	Visibility *IngressVisibility `json:"visibility,omitempty"`
+	Visibility IngressVisibility `json:"visibility,omitempty"`
 }
 
 // IngressVisibility describes whether the Ingress should be exposed to
 // public gateways or not.
-type IngressVisibility struct {
-	// True iff the ClusterIngress should not be exposed to public gateways.
-	LocalOnly bool `json:"localOnly,omitempty"`
-}
+type IngressVisibility string
+
+const (
+	IngressVisibilityClusterLocal IngressVisibility = "ClusterLocal"
```

Comment:
What other values can this take on?

/lint

Reply:
Added IngressVisibilityExternalIP, and default to that.

The reason for defaulting to having external IP is because "Ingress" usually means that.